### DISCUSSION TITLE
How to put condition code in .ps1 files?

### DISCUSSION THREAD

**Discussion by StevenTCramer** [dotnet/templating#5653] [Template authoring] [answered]
I have read https://github.com/dotnet/templating/wiki/Conditional-processing-and-comment-syntax#other-file-types

I am trying to conditionally include code based on the boolean `web` parameter.  But when 'web' is set to false the code is still being included.

```powershell
$Env:ASPNETCORE_ENVIRONMENT = "Development"

Push-Location $PSScriptRoot
try {
  #if(web)
  Push-Location ./Source/ContainerApps/Web/Web.TypeScript
  npm install  
  Pop-Location
  #endif

  Push-Location DevOps/Tye
  tye run --dashboard --watch --logs console -v Debug
  Pop-Location
}
finally {
  Pop-Location
}
```

**Comment by GangWang01** [comment, marked as answer]
`.ps1` file is the case [Other File Types](https://github.com/dotnet/templating/blob/main/docs/Conditional-processing-and-comment-syntax.md#other-file-types). Note that the conditional expression should be prefixed with `//` like below.
```
  //#if(web)
  Push-Location ./Source/ContainerApps/Web/Web.TypeScript
  npm install  
  Pop-Location
  //#endif
```

**Reply by StevenTCramer** [reply]
@GangWang01  The comment in Powershell is # not // so the above is not valid powershell.

**Reply by GangWang01** [reply]
> @GangWang01 The comment in Powershell is # not // so the above is not valid powershell.

`//` is used by Template Engine working together with conditional expression for `.ps1` file in the template. 
Template Engine removes `//#if(web)` and `//#endif` in the template `.ps1` file when instantiating the template. The output `.ps1` file will not have `//` included.

Let me know if  it doesn't work after your try.

**Reply by vlada-shubina** [reply]
Suggestion provided by @GangWang01 is very valid.

However, I think we should also consider adding powershell (*.ps1) in [single hash comment group](https://github.com/dotnet/templating/blob/main/docs/Conditional-processing-and-comment-syntax.md#single-hash-line-comments) so even when conditions are used the file remains correct Powershell.

**Reply by StevenTCramer** [reply]
When using the `//` the generated template is correct but now I can no longer use the ps1 script in the template source project.  
The goal of dotnet templates is to have them run-able and testable.  To accomplish that I think ps1 files do need to be moved into the single hash comment group.

Thanks much for the feedback.
